### DISCUSSION TITLE
Socket log spam

### DISCUSSION THREAD

**Discussion by naanlizard** [OvenMediaLabs/OvenMediaEngine#1511] [Help]
Hello all

In our logging setup we are getting a seriously large amount of spam in the logs that is basically all the following warning, hundreds of times a second.

>[2024-01-27 12:40:59.657] W [StreamWorker:848855] Socket | socket.cpp:1058 | [#69] [0x7f6b7d004c10] IsSendable(): Invalid state: Disconnected (expected: == SocketState::Connected) - <ClientSocket: 0x7f6b7d004c10, #69, Disconnected, TCP, Nonblocking, x.x.x.x:34874>

Things seem to be running fine, so I'm not sure what's up with that. Is it possibly something we're doing in our frontend?

Another related error that crops up, but less commonly

2024-01-27 12:12:49 | [2024-01-27 12:12:49.497] E [SPICE-t3478:21] Socket \| socket.cpp:1627 \| [#125] [0x7f6b7d004010] An error occurred while read data: [Socket] [errno] Connection timed out (110) (110)
  |   |   | 2024-01-27 12:12:49 | Stack trace: #0   /opt/ovenmediaengine/bin/OvenMediaEngine 0x55a97672dd7a ? + 0x48ed7a
  |   |   | 2024-01-27 12:12:49 | #1   /opt/ovenmediaengine/bin/OvenMediaEngine 0x55a97672e192 ? + 0x48f192
  |   |   | 2024-01-27 12:12:49 | #2   /opt/ovenmediaengine/bin/OvenMediaEngine 0x55a976733c60 ? + 0x494c60
  |   |   | 2024-01-27 12:12:49 | #3   /opt/ovenmediaengine/bin/OvenMediaEngine 0x55a97673dcfb ? + 0x49ecfb
  |   |   | 2024-01-27 12:12:49 | #4   /lib/x86_64-linux-gnu/libstdc++.so.6 0x7f6b89addde4 ? + 0xd6de4
  |   |   | 2024-01-27 12:12:49 | #5   /lib/x86_64-linux-gnu/libpthread.so.0 0x7f6b8ad97609 ? + 0x9609
  |   |   | 2024-01-27 12:12:49 | #6   /lib/x86_64-linux-gnu/libc.so.6     0x7f6b897cd293 clone + 0x43
  |   |   | 2024-01-27 12:12:49 |  

Any thoughts are appreciated, it makes debugging other things difficult and makes us rotate logs more than we'd like

**Comment by bchah** [comment]
You should be able to suppress it via Logger.xml:

`<Tag name="Socket" level="critical" />`

**Comment by dimiden** [comment]
This log allows OME developers to analyze when unexpected behavior occurs. Since this is code included at the time of initial development, there may be some parts that are somewhat inconsistent with the current version. I need more information to investigate this, so could you tell me which OME version the above log message is displayed on and which OS version is running? I would appreciate it if you could also attach the log.

**Reply by naanlizard** [reply]
Ubuntu 22.04.3 LTS
Kernel 6.5.0-15-generic
OME version: docker latest image, pinned to a specific commit which I'm not sure how to find.

`FROM airensoft/ovenmediaengine@sha256:167de56e8b6600e0f459038780fff17659d7f835a3464834c20403b1e694564e`

Same version as reported here https://github.com/AirenSoft/OvenMediaEngine/issues/1445 which is apparently 16.0, and no later than November 7, 2023 at 7:09:24 AM GMT according to our commits

I *assume* but am not sure it's this commit https://github.com/AirenSoft/OvenMediaEngine/commit/c8bb321c551c65b7e3583e7556f58e71877c1cbe

[logs-20240211194357.tar.gz](https://github.com/AirenSoft/OvenMediaEngine/files/14233780/logs-20240211194357.tar.gz)

**Reply by dimiden** [reply]
I'm analyzing the issue based on your information. However, as it may take a significant amount of time to analyze, if immediate action is necessary, as suggested by @bchah, temporarily set the log level to error or critical to suppress the warning messages.
